src/backend/cdb/cdbpullup.c

```diff
@@ -305,10 +307,36 @@ cdbpullup_findEclassInTargetList(EquivalenceClass *eclass, List *targetlist,
 		Expr	   *key = (Expr *) em->em_expr;
 		ListCell *lc_tle;
 
-		if (OidIsValid(hashOpFamily) &&
-			!get_opfamily_member(hashOpFamily, em->em_datatype, em->em_datatype,
-								 HTEqualStrategyNumber))
-			continue;
+		if (OidIsValid(hashOpFamily))
+		{
+			/*
+			 * To check if the member is hashable, we should invoke cdb_hashproc_in_opfamily
+			 * to try to see if there is a hash proc for the type in the opfamily.
+			 *
+			 * Previous code just get_opfamily_member here, that is not enough, since
+			 * there are binary compatible types, e.g. varchar and text.
+			 *
+			 * See Issue: https://github.com/greenplum-db/gpdb/issues/12700 for a detailed case.
+			 *
+			 * Since cdb_hashproc_in_opfamily might raise error, so we enclose the call by PG_TRY.
+			 * Error means we find nothing. Auto Variable hash_proc (without volatile in declartion)
+			 *  is set in both Try clause and Catch clause.
```

Comment:
```suggestion
			 * is set in both Try clause and Catch clause.
```
Nit: suspicous whitespace.

Reply:
Done, thanks.